app/services/record_declarations/base.rb

```diff
@@ -86,6 +89,17 @@ def find_or_create_record!
       end
     end
 
+    def record_exist?
+      declaration = self.class.declaration_model.find_by(
+        user: user,
+        course_identifier: course_identifier,
+        declaration_type: declaration_type,
+        cpd_lead_provider: cpd_lead_provider,
+      )
```

Comment:
I would say the cpd_lead_provider does not need to be there either - if user switches LPs, and has a started declaration, we should prevent the new LP from submitting another started declaration.